wcfsetup/install/files/lib/acp/form/StyleAddForm.class.php

```diff
@@ -768,6 +797,15 @@ public function save() {
 			'styleDescription' => 'wcf.style.styleDescription'.$style->styleID
 		]);
 		
+		// save compiled style
+		if ($this->styleTestFileDir && file_exists($this->styleTestFileDir . '/style.css') && file_exists($this->styleTestFileDir . '/style-rtl.css')) {
+			$styleFilename = StyleCompiler::getFilenameForStyle($style);
+			rename($this->styleTestFileDir . '/style.css', $styleFilename . '.css');
+			rename($this->styleTestFileDir . '/style-rtl.css', $styleFilename . '-rtl.css');
+			
+			rmdir($this->styleTestFileDir);
+		}
```

Comment:
The CSS must not be written in StyleAddForm, because the imagePath is not yet known.